app/bucketlists/views.py

```diff
@@ -72,23 +76,54 @@ def create_new_bucketlist():
 @auth.login_required
 def fetch_all_bucketlists():
     """Returns all bucketlists"""
-    bucketlists = BucketList.query.all()
-    for bucketlist in bucketlists:
-        items = BucketListItem.query.filter_by(
+
+    page_no = request.args.get('pag_no', 1)
+    limit = request.args.get('limit', 20)
+    q_name = request.args.get('q', "")
+
+    bucketlists = BucketList.query.filter_by(
+        created_by=g.user_id).filter(
+        BucketList.name.like('%{}%.'.format(q_name))).paginate(
+        int(page_no), int(limit))
+
+    if not bucketlists:
+        abort(404, {"error": "No bucketlists added"})
+    for bucketlist in bucketlists.items:
+        bucket_items = BucketListItem.query.filter_by(
             bucketlist_id=bucketlist.id).first()
-        if items is None:
-            items = []
+        if bucket_items is None:
+            bucket_items = []
         else:
-            items = items.name
-        buckets.append({'id': bucketlist.id,
-                        'name': bucketlist.name,
-                        'items': items,
-                        'date_created': bucketlist.date_created,
-                        'date_modified': bucketlist.date_modified
-                        })
-    if not len(buckets):
-        return jsonify({"error": "No bucketlists added"}), 404
-    return jsonify({'bucketlists': buckets}), 200
+            list_items = [bucket_item.name for bucket_item in bucket_items]
+        # buckets.append({'id': bucketlist.id,
+        #                 'name': bucketlist.name,
+        #                 'items': bucket_items,
+        #                 'date_created': bucketlist.date_created,
+        #                 'date_modified': bucketlist.date_modified
+        #                 })
```

Comment:
You probably have a reason having the above code commented out. I think should delete it utilize git incase you need the code back. What do you think?